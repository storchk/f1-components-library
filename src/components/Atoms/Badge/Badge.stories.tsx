import type { Meta, StoryObj } from '@storybook/react'

import { Badge } from './Badge'
import type { BadgeProps } from './Badge.types'

export default {
  title: 'Atoms/Badge',
  component: Badge,
  args: {
    label: 'Badge',
  },
} as Meta

type Story = StoryObj<BadgeProps>

export const Primary: Story = {
  render: props => <Badge variant="primary" {...props} />,
}
export const Secondary: Story = {
  render: props => <Badge variant="secondary" {...props} />,
}
