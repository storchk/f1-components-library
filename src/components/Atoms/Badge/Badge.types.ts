import type { ThemeType } from '../../../theme'

export type BadgeVariant = 'primary' | 'secondary'

export type BadgeProps = {
  variant?: BadgeVariant
  label: string
}

export type StyledBadgeProps = {
  $variant: BadgeVariant
}

export type GetBadgeVariantsProps = {
  theme: ThemeType
} & StyledBadgeProps
