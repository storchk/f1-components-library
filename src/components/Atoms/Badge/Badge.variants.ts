import { css } from 'styled-components'

import type { GetBadgeVariantsProps } from './Badge.types'

export const getBadgeVariants = ({ theme, $variant }: GetBadgeVariantsProps) => {
  switch ($variant) {
    case 'primary':
    default:
      return css`
        color: ${theme.colors.primary};
        border-color: ${theme.colors.primary};
      `
    case 'secondary':
      return css`
        color: ${theme.colors.secondary};
        padding: ${theme.spacing.xxs} ${theme.spacing.xs};
        border-radius: ${theme.size.md}px;
        border: ${theme.border.width.small} solid ${theme.colors.primary};
      `
  }
}
