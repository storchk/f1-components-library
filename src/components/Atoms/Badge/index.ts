export { Badge } from './Badge'
export type { BadgeProps, BadgeVariant } from './Badge.types'
