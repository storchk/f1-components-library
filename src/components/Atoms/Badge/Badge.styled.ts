import styled, { css } from 'styled-components'

import type { StyledBadgeProps } from './Badge.types'
import { getBadgeVariants } from './Badge.variants'

export const StyledBadge = styled.span<StyledBadgeProps>(
  ({ theme, ...props }) => css`
    display: inline-flex;
    justify-content: center;
    align-items: center;
    ${getBadgeVariants({ theme, ...props })}
  `
)
