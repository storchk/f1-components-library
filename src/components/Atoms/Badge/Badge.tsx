import { Typography } from '../Typography'
import { StyledBadge } from './Badge.styled'
import type { BadgeProps } from './Badge.types'
export const Badge = ({ variant = 'primary', label, ...props }: BadgeProps): JSX.Element => {
  return (
    <StyledBadge $variant={variant} {...props}>
      <Typography $isUppercase fontSize="xxs" fontWeight="semibold">
        {label}
      </Typography>
    </StyledBadge>
  )
}
