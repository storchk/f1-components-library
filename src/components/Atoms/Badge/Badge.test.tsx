import { testing } from '../../../testing'
import { Badge } from './Badge'

const { render, screen, renderer } = testing
const label = "Hello world! I am using React"

describe('Atoms: Badge', () => {
  it('should render badge with text', () => {
    render( <Badge label={label} />)
    const text = screen.getByText(/Hello world! I am using React/i)
    expect(text).toBeInTheDocument()
  })
  it('snapshot | variant: primary', () => {
    const view = renderer(<Badge variant="primary" label={label} />)
    expect(view).toMatchSnapshot()
  })
  it('snapshot | variant: secondary', () => {
    const view = renderer(<Badge variant="secondary" label={label} />)
    expect(view).toMatchSnapshot()
  })
})
