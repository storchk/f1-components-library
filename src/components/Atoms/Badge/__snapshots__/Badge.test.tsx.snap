// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms: Badge snapshot | variant: primary 1`] = `
.c1 {
  font-size: 12px;
  font-weight: 600;
  color: inherit;
  font-family: Titillium Web;
  text-transform: uppercase;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #e10600;
  border-color: #e10600;
}

<span
  className="c0"
>
  <span
    className="c1"
    fontSize="xxs"
    fontWeight="semibold"
  >
    Hello world! I am using React
  </span>
</span>
`;

exports[`Atoms: Badge snapshot | variant: secondary 1`] = `
.c1 {
  font-size: 12px;
  font-weight: 600;
  color: inherit;
  font-family: Titillium Web;
  text-transform: uppercase;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #1f1f27;
  padding: 4px 8px;
  border-radius: 24px;
  border: 2px solid #e10600;
}

<span
  className="c0"
>
  <span
    className="c1"
    fontSize="xxs"
    fontWeight="semibold"
  >
    Hello world! I am using React
  </span>
</span>
`;
